soc/sifive/sifive_freedom/e300/Kconfig.soc

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2024 Antmicro <www.antmicro.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_SIFIVE_FREEDOM_E300
+	bool
+	select SOC_FAMILY_SIFIVE_FREEDOM
+
+config SOC_SERIES
+	default "e300" if SOC_SERIES_SIFIVE_FREEDOM_E300
```

Comment:
We are planning to have the value and name of this linked in future: https://github.com/zephyrproject-rtos/zephyr/issues/69317
But this raises an interesting issue @tejlmand because if just "e300" is used, that's so generic it's bound to have a name clash in future

Reply:
It's a future thing, e300 is fine for now but yes in future might have to expand it

Reply:
> It's a future thing, e300 is fine for now but yes in future might have to expand it

I think current proposal is fine.
If we're later going to use those values internally, like we do the the `CONFIG_SOC`, then we should be generating those.
Just like we do for boards.

But there are a couple of reasons this is not done in first iteration of hwmv2, so let's not worry to much now.
But thanks for raising awareness on this @nordicjm 👍